AGENTS_MIGRATION.md

```diff
@@ -0,0 +1,164 @@
+
+### 3. GameMasterAgentFactory
+- Creates properly configured `ChatCompletionAgent` instances
+- Combines adventure context, character information, and system prompts
+- Centralizes agent configuration logic
+
+## Extension Points for Multi-Agent Support
+
+### 1. Agent Group Coordination (AgentChatClient.cs)
+```csharp
+// TODO: Future enhancement - Create AgentGroupChat for multi-agent scenarios:
+// var agentGroup = new AgentGroupChat(GameMasterAgent, npcAgent1, npcAgent2);
+// await foreach (var message in agentGroup.InvokeAsync(history))
+```
+
+**Implementation Strategy:**
+- Create an `AgentGroupChat` that manages conversation turns between agents
+- Define agent roles (GameMaster, NPC, WorldDescriptor, etc.)
+- Implement turn-taking logic based on conversation context
+
+### 2. Multiple Agent Types (ServiceExtensions.cs)
+```csharp
+// EXTENSION POINT: Future multi-agent support could register additional agents here
+// For example:
+// services.AddTransient<NPCAgent>(sp => CreateNPCAgent(sp, "Merchant"));
+// services.AddTransient<WorldAgent>(sp => CreateWorldAgent(sp));
+```
+
+**Suggested Agent Types:**
+- **NPCAgent**: Roleplay specific non-player characters
+- **WorldAgent**: Provide environmental descriptions and world state
+- **RulesAgent**: Handle rule enforcement and dice rolling
+- **NarratorAgent**: Focus on storytelling and scene setting
+
+### 3. Agent Configuration (GameMasterAgentFactory.cs)
+```csharp
+// EXTENSION POINT: Agent configuration could be enhanced for multi-agent scenarios
+// For example, different agent types could have different configurations:
+// - GameMasterAgent: Narrative control, rule enforcement
+// - NPCAgent: Character roleplay for specific NPCs
+// - WorldAgent: Environmental descriptions and world state
+```
+
+## Implementation Example: Adding NPC Agents
+
+### Step 1: Create NPC Agent Factory
+```csharp
+public class NPCAgentFactory
+{
+    public ChatCompletionAgent CreateNPCAgent(Character npc, Adventure adventure, Kernel kernel)
+    {
+        return new ChatCompletionAgent
+        {
+            Name = $"NPC_{npc.Name}",
+            Description = $"Roleplay agent for {npc.Name}",
+            Instructions = $"You are {npc.Name}, a {npc.Specialization}. {npc.CharacterSheet}",
+            Kernel = kernel
+        };
+    }
+}
+```
+
+### Step 2: Register Multiple Agents
+```csharp
+services.AddTransient<NPCAgentFactory>();
+services.AddTransient<IEnumerable<Agent>>(sp =>
+{
+    var adventure = sp.GetRequiredService<Adventure>();
+    var kernel = sp.GetRequiredService<Kernel>();
+    var npcFactory = sp.GetRequiredService<NPCAgentFactory>();
+    var gmFactory = sp.GetRequiredService<GameMasterAgentFactory>();
+    
+    var agents = new List<Agent>
+    {
+        gmFactory.CreateGameMasterAgent(adventure, character, kernel)
+    };
+    
+    // Add NPC agents for each character in the adventure
+    agents.AddRange(adventure.Characters.Select(npc => 
```

Comment:
You misunderstand the point of this application. There will be one player only, but agents will help the system respond to it in complex manners while keeping its response fair. For example, if the player tries to cast something they shouldn't be able to cast, we tell them they can't do that. Additionally, it should make sure the game master isn't trying to take actions on the player's behalf and it should nudge the adventure forward when a path forward isn't clear.